[DOCS] Advanced rule query preview option available (#2251)
elastic/security-docs: docs/detections/rules-ui-create.asciidoc

To preview a rule:

. Write the rule query.
. Choose how you want to preview the query results:

> ** *Quick query preview*: Allows you to select from pre-defined timeframes -- *Last hour*, *Last day*, or *Last month* -- when previewing rule results. Note that threshold and event correlation rules have limited timeframe options. The rule interval and look-back time are also pre-defined for the preview and differ by rule type. These settings cannot be modified.
> ** *Advanced query preview*: Allows you to choose a custom timeframe for the rule preview, schedule how often the rule should run, and specify a look-back time.

Reviewer: Minor edit suggestion to reduce repetition and filler. Also, looks like two-word "time frame" is more prevalent in our docs.
```suggestion
** *Quick query preview*: Select from pre-defined time frames -- *Last hour*, *Last day*, or *Last month* -- when previewing rule results. Note that threshold and event correlation rules have limited time frame options. The rule interval and look-back time are also pre-defined for the preview and differ by rule type. These settings cannot be modified.
** *Advanced query preview*: Choose a custom time frame for the rule preview, schedule how often the rule should run, and specify a look-back time.
```

+

> TIP: Avoid setting long timeframes with short rule intervals. This might cause the rule preview to timeout.

Reviewer: ```suggestion
TIP: Avoid setting long time frames with short rule intervals. This might cause the rule preview to timeout.
```

> TIP: It is recommended to set the `Additional look-back time` to at least 1 minute. This ensures there are no missing alerts when a rule does not run exactly at its scheduled time. In addition, avoid setting long timeframes with short rule intervals. This might cause the rule preview to timeout.

Reviewer 2: ```suggestion
TIP: Avoid setting long timeframes with short rule intervals. This might cause the rule preview to timeout.
```
In preview, alerts won't be missed due to scheduling drift so the additional lookback time is not strictly necessary. It's there for users to select so they can match the real desired parameters as closely as possible, and it still has an effect on how the rule executes, but we can leave out the recommendation here IMO.

To preview a rule:

. Write the rule query.
. Choose how you want to preview the query results:

> ** *Quick query preview*: Allows you to select from three pre-defined timeframes -- *Last hour*, *Last day*, or *Last month* -- when previewing rule results. By default, rules are scheduled to run every 5 minutes with an additional look-back time of 1 minute. These settings cannot be modified.

Reviewer 3: Looks great, though the default timeframes and values are a little bit more complicated.

The pre-defined timeframes are different for different rule types:

- Custom Query: hour, day, month
- Machine Learning: hour, day, month
- Threshold: hour
- Event Correlation: hour, day
- Indicator Match: hour, day, week
- New Terms: hour, day, month

Also, default "scheduled to run" and "additional look-back time" values are different for different timeframes.

- "Last hour": "scheduled to run" set to 5 minutes and "additional look-back time" set to 1 minute
- "Last day": "scheduled to run" set to 1 hour and "additional look-back time" set to 5 minutes
- "Last week": "scheduled to run" set to 1 hour and "additional look-back time" set to 5 minutes
- "Last month": "scheduled to run" set to 1 day and "additional look-back time" set to 1 hour

Author: Ah, I remember that the pre-defined timeframes are different for certain rule types but didn't know about the rule interval and look-back time values. Thanks for sharing this information! I'll work on integrating these next week when I'm back from PTO.

> ** *Advanced query preview*: Allows you to choose a custom timeframe for the rule preview, schedule how often the rule should run, and specify a look-back time.

Reviewer 3: We could mention that if user selects short rule interval while the timeframe is set to a long period, the warning message will appear indicating about potential preview execution timeout.

<img width="942" alt="Screenshot 2022-08-04 at 18 16 35" src="https://user-images.githubusercontent.com/2700761/182898056-27e53b63-ca98-4987-a9d9-8915909a7738.png">

Author: I incorporated this information as a tip under this step. Lemme know what you think when you have a chance @e40pud  - thanks!
